tests/foreman/api/test_computeresource_gce.py

```diff
@@ -0,0 +1,250 @@
+
+        :CaseLevel: Integration
+        """
+        assert len(GCEnvironment.gce_compute().available_flavors()) > 1
+
+    @tier3
+    def test_positive_create_finish_template_image(self):
+        """Finish template image along with username is being added in GCE CR
+
+        :id:
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a finish template based image in GCE CR.
+
+        :expectedresults: Finish template image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert GCEnvironment.finishimg(
+            gceclient).compute_resource.id == GCEnvironment.gce_compute().id
+        assert GCEnvironment.finishimg(gceclient).uuid == GCEnvironment.latest_rhel_uuid()
+        assert GCEnvironment.finishimg().username == GCEnvironment.finishimg_user
+
+    @tier3
+    def test_positive_create_cloud_init_image(self):
+        """Cloud Init template image along with username is being added in GCE CR
+
+        :id:
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a cloud init template in GCE CR.
+
+        :expectedresults: Cloud init image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert GCEnvironment.cloudimg(
+            gceclient).compute_resource.id == GCEnvironment.gce_compute().id
+        assert GCEnvironment.cloudimg(gceclient).uuid == GCEnvironment.custom_cloudinit_uuid()
+        assert GCEnvironment.cloudimg().username == GCEnvironment.cloudimg_user
+
+
+class GCEHostProvisioningTestCase(APITestCase):
+    """
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+
+        :return:
+        """
+        super(GCEHostProvisioningTestCase, cls).setUpClass()
+        # Provision a Host and revolve all the tests around it
+        cls.mtype = 'g1-small'
+        cls.network = 'default'
+        cls.volsize = '13'
+        cls.hostname = gen_string('alpha')
+
+        cls.compute_attrs = {
+            'image_id': GCEnvironment.latest_rhel_uuid(gceclient),
+            'machine_type': cls.mtype,
+            'network': cls.network,
+            'associate_external_ip': True,
+            'volumes_attributes': {
+                '0': {'size_gb': cls.volsize}
+            }
+        }
+        cls.host = entities.Host(
+            architecture=GCEnvironment.architecture(), compute_attributes=cls.compute_attrs,
+            domain=GCEnvironment.domain(), hostgroup=GCEnvironment.hostgroup(gceclient),
+            image=GCEnvironment.finishimg(gceclient), organization=GCEnvironment.organization(),
+            operatingsystem=GCEnvironment.operatingsystem(), location=GCEnvironment.location(),
+            name=cls.hostname, provision_method='image', root_pass=gen_string('alphanumeric')
+        ).create()
+        cls.fullhostname = '{}.{}'.format(cls.hostname, GCEnvironment.domain().name).lower()
```

Comment:
That should be specified in relevant tests, not in class setup. Setup is reserved for time-consuming operations where results can be used by multiple tests.

Reply:
Probably not in current tests, but it will be used multiple times in future.